### PR TITLE
boards: mimxrt1050_evk: EVK to EVKB documentation update

### PR DESCRIPTION
**Pull request by agansari**
Updated documentation for i.MX RT EVK board.
EVK was discontinued and replaced by EVKB board revision.

### CODE REVIEW

File: boards/arm/mimxrt1050_evk/doc/mimxrt1050_evk.rst [hunk 1]
```diff
@@ -23,7 +26,7 @@ basic 2D graphics, camera interface, SPDIF, and I2S audio interface.
 Hardware
 ********
 
-- MIMXRT1052DVL6A MCU (600 MHz, 512 KB TCM)
+- MIMXRT1052DVL6B MCU (600 MHz, 512 KB TCM)
```
Comment (MaureenHelm) [inline boards/arm/mimxrt1050_evk/doc/mimxrt1050_evk.rst]: We still need to add the A1 silicon part numbers to https://github.com/zephyrproject-rtos/zephyr/blob/55b462cdfa8dcda620dc1cc984137ee1887e92a7/soc/arm/nxp_imx/rt/Kconfig.soc#L54-L59

But that depends on the next MCUXpresso SDK release.

Until then, let's keep this document to be about A0 silicon and EVK board, but make a note that the A0/EVK Zephyr configuration also works for A1/EVKB.

**Review by MaureenHelm — CHANGES_REQUESTED** [review]

File: boards/arm/mimxrt1050_evk/doc/mimxrt1050_evk.rst (hunk 1 above)
Comment (jhqian) [reply boards/arm/mimxrt1050_evk/doc/mimxrt1050_evk.rst]: agree, @MaureenHelm , the part numbers will be updated after the next SDK mainline release.

**Comment by dbkinder** [comment]
Check https://app.shippable.com/github/zephyrproject-rtos/zephyr/runs/23028/5/tests for commit message issues

File: boards/arm/mimxrt1050_evk/doc/mimxrt1050_evk.rst [hunk 2]
```diff
@@ -15,6 +15,9 @@ Bluetooth™, GPS, displays, and camera sensors. As with other i.MX processors,
 i.MX RT1050 also has rich audio and video features, including LCD display,
 basic 2D graphics, camera interface, SPDIF, and I2S audio interface.
 
+The following document refers to MIMXRT1050-EVKB board, for the discontinued
```
Comment (dbkinder) [inline boards/arm/mimxrt1050_evk/doc/mimxrt1050_evk.rst]: Better would be:

    This document describes support for the MIMXRT1050-EVKB board.
    For the discontinued MIMXRT1050-EVK board, see the `Board Revisions`_ section.

File: boards/arm/mimxrt1050_evk/doc/mimxrt1050_evk.rst [hunk 3]
```diff
@@ -182,12 +186,25 @@ your Zephyr application to flash. It will leave you at a GDB prompt.
    :board: mimxrt1050_evk
    :goals: debug
 
+Board Revisions
+***************
+
+MIMXRT1050-EVKB is revison A1, while MIMXRT1050-EVK board is rev. A0. There are
```
Comment (dbkinder) [inline boards/arm/mimxrt1050_evk/doc/mimxrt1050_evk.rst]: Better would be:

    The original MIMXRT1050-EVK (rev A0) board was updated with a newer
    MIMXRT1050-EVKB (rev A1) board, with these major hardware differences:

File: boards/arm/mimxrt1050_evk/doc/mimxrt1050_evk.rst [hunk 4]
```diff
@@ -182,12 +186,25 @@ your Zephyr application to flash. It will leave you at a GDB prompt.
...
+the following major hardware differences:
+
+- SoC changed from MIMXRT1052DVL6**A** to MIMXRT1052DVL6**B**
+- Hardware bug fixes for: power, interfaces, memory
```
Comment (dbkinder) [inline boards/arm/mimxrt1050_evk/doc/mimxrt1050_evk.rst]: change to:   ``power, interfaces, and memory``

File: boards/arm/mimxrt1050_evk/doc/mimxrt1050_evk.rst [hunk 5]
```diff
@@ -206,3 +223,6 @@ your Zephyr application to flash. It will leave you at a GDB prompt.
 
 .. _Segger J-Link OpenSDA V2.1 Firmware:
    https://www.segger.com/downloads/jlink/OpenSDA_V2_1.bin
+
+.. _application note:
```
Comment (dbkinder) [inline boards/arm/mimxrt1050_evk/doc/mimxrt1050_evk.rst]: and update this name too:

    .. _NXP i.MXRT1050 A0 to A1 Migration Guide:

File: boards/arm/mimxrt1050_evk/doc/mimxrt1050_evk.rst [hunk 6]
```diff
@@ -182,12 +186,25 @@ your Zephyr application to flash. It will leave you at a GDB prompt.
...
+***************
+
+MIMXRT1050-EVKB is revison A1, while MIMXRT1050-EVK board is rev. A0. There are
+the following major hardware differences:
+
+- SoC changed from MIMXRT1052DVL6**A** to MIMXRT1052DVL6**B**
+- Hardware bug fixes for: power, interfaces, memory
+- Arduino headers included
+
+For more details, please see the following `application note`_.
```
Comment (dbkinder) [inline boards/arm/mimxrt1050_evk/doc/mimxrt1050_evk.rst]: Change to:

    For more details, please see this `NXP i.MXRT1050 A0 to A1 Migration Guide`_,

**Review by dbkinder — CHANGES_REQUESTED** [review]

File: boards/arm/mimxrt1050_evk/doc/mimxrt1050_evk.rst [hunk 7]
```diff
@@ -15,6 +15,9 @@ Bluetooth™, GPS, displays, and camera sensors. As with other i.MX processors,
 i.MX RT1050 also has rich audio and video features, including LCD display,
 basic 2D graphics, camera interface, SPDIF, and I2S audio interface.
 
+The following document refers to the discontinued MIMXRT1050-EVK board, for the
+MIMXRT1050-EVKB board, refer to `Board Revisions`_ section.
```
Comment (MaureenHelm) [inline boards/arm/mimxrt1050_evk/doc/mimxrt1050_evk.rst]: I think you need to add a link target in the board revisions section for this link to work. @dbkinder?

Comment (dbkinder) [reply boards/arm/mimxrt1050_evk/doc/mimxrt1050_evk.rst]: This link refers to the (implicit) target for the section title "Board Revisions" found later in this document.  (If you were referring to a section title in another document, you'd need to add an explicit target.)  So, this reference is OK as is.

File: boards/arm/mimxrt1050_evk/doc/mimxrt1050_evk.rst [hunk 8]
```diff
@@ -15,6 +15,9 @@ Bluetooth™, GPS, displays, and camera sensors. As with other i.MX processors,
 i.MX RT1050 also has rich audio and video features, including LCD display,
 basic 2D graphics, camera interface, SPDIF, and I2S audio interface.
 
+The following document refers to the discontinued MIMXRT1050-EVK board, for the
```
Comment (dbkinder) [inline boards/arm/mimxrt1050_evk/doc/mimxrt1050_evk.rst]: change the comma to a period and start a new sentence:

     This document refers to the discontinued MIMRT1050-EVK board.  For the

File: boards/arm/mimxrt1050_evk/doc/mimxrt1050_evk.rst [hunk 9]
```diff
@@ -187,12 +191,25 @@ your Zephyr application to flash. It will leave you at a GDB prompt.
...
    :goals: debug
 
+Board Revisions
+***************
+
+The original MIMXRT1050-EVK (rev A0) board was updated with a newer
+MIMXRT1050-EVKB (rev A1) board, with these major hardware differences::
+
+- SoC changed from MIMXRT1052DVL6**A** to MIMXRT1052DVL6**B**
+- Hardware bug fixes for: power, interfaces and memory
```
Comment (dbkinder) [inline boards/arm/mimxrt1050_evk/doc/mimxrt1050_evk.rst]: add a comma after interfaces:

    power, interfaces, and memory

**Review by dbkinder — DISMISSED** [review]
almost there...

**Comment by agansari** [comment]
@dbkinder I've made the changes you requested.

File: boards/arm/mimxrt1050_evk/doc/mimxrt1050_evk.rst (hunk 7 above)
Comment (MaureenHelm) [reply boards/arm/mimxrt1050_evk/doc/mimxrt1050_evk.rst]: Ok, thanks

**Review by MaureenHelm — APPROVED** [review]

**Pull request merged**